### PR TITLE
ETQ Instructeur, lorsque une demande en retour est soumise à nouveau, je reçois un mail « Vous avez un retour sur votre demande de changement » plutôt que « Vous avec une nouvelle demande »

### PR DESCRIPTION
**Pull request by Isalafont**
enrollment_mailer : new method to send email to instructor when demandeur submit enrollment after changes_requested

Add some test to validate email subject and email text.
Trigger email sent before transition from changes_requested to submitted in enrollment models. 

Need to implement test in `change_state_spec.rb`
+ some refacto.

### CODE REVIEW

**Comment by Isalafont** [comment]
[ETQ Instructeur, lorsque une demande en retour est soumise à nouveau, je reçois un mail « Vous avez un retour sur votre demande de changement » plutôt que « Vous avec une nouvelle demande »](https://trello.com/c/Gv1meZCP/934-etq-instructeur-lorsque-une-demande-en-retour-est-soumise-%C3%A0-nouveau-je-re%C3%A7ois-un-mail-vous-avez-un-retour-sur-votre-demande-de-c)

**Review by rdubigny — CHANGES_REQUESTED** [review]
Without deep diving into this PR, it misses some code removal. I like the implementation you propose. I'm ok to go with this in production. But before that you should handle the case of a user submiting his pass from a changes_requested state, the instructor will then receive 2 mails instead of one.

**Comment by Isalafont** [comment]
@rdubigny Thanks ! 
I still have some work to do. I would like to implement more test in `change_resquest_spec.rb` to check that : 
- when enrollment pass from draft to submitted, the `notify_submited` with subject "Vous avez une nouvelle demande d'habiliation" is sent. 
- when enrollment pass from `change_request` to `submitted` the notify_submitted with subject "Vous avez une retour sur une demande d'habilitation" is sent. 
I was thinking about adding a `non regression`test to check sidekiq too. 

then I will do some refacto, adding a new method to trigger the first `notify_sumitted` and could remove code from `Data_provider_config.yml`

File: backend/config/data_providers.yml [hunk 1]
```diff
@@ -11,8 +11,6 @@ shared:
       subject: "Votre habilitation a été refusée"
     revoke: &shared_revoke_mailer
       subject: "Votre habilitation a été révoquée"
-    notify_submitted: &shared_notify_submitted_mailer
```
Comment (rdubigny) [inline backend/config/data_providers.yml:14]: you should remove this event from the doc also : at backend/app/views/enrollment_mailer/README.md

File: backend/app/mailers/enrollment_mailer.rb [hunk 2]
```diff
@@ -51,6 +51,36 @@ def notification_email_to_instructors
...
+  def notify_instructors_submitted_changes_requested
+    @target_api_label = data_provider_config["label"]
+    @enrollment = Enrollment.find(params[:enrollment_id])
+    @demandeur_email = @enrollment.demandeurs.pluck(:email).first
+    @url = "#{ENV.fetch("FRONT_HOST")}/#{params[:target_api].tr("_", "-")}/#{params[:enrollment_id]}"
+
+    mail(
+      to: @enrollment.subscribers.pluck(:email),
+      from: "notifications@api.gouv.fr",
+      subject: "Vous avez un retour sur votre demande de changement",
```
Comment (rdubigny) [inline backend/app/mailers/enrollment_mailer.rb:78]: I'd rather keep the imperative form for more consistence and conciseness. Plus, I'd keep the same wording everywere for this action "demande de changement" => "demande de modification".

```suggestion
      subject: "Retour sur votre demande de modification",
```

File: backend/app/models/enrollment.rb [hunk 3]
```diff
@@ -304,6 +312,22 @@ def notify_administrators_for_unknown_software_enrollment
     ).notification_email_unknown_software.deliver_later
   end
 
+  def notify_subscribers_for_new_submitted_enrollment
+    EnrollmentMailer.with(
+      target_api: target_api,
+      enrollment_id: id,
+      demandeur_email: demandeurs.pluck(:email).first
+    ).notify_instructors_submitted_enrollment.deliver_later
```
Comment (rdubigny) [inline backend/app/models/enrollment.rb:320]:
```suggestion
    ).new_enrollment_submission_notification_email.deliver_later
```

File: backend/app/models/enrollment.rb [hunk 4]
```diff
@@ -304,6 +312,22 @@ def notify_administrators_for_unknown_software_enrollment
     ).notification_email_unknown_software.deliver_later
   end
 
+  def notify_subscribers_for_new_submitted_enrollment
```
Comment (rdubigny) [inline backend/app/models/enrollment.rb:315]:
```suggestion
  def notify_subscribers_for_new_enrollment_submission
```

File: backend/app/models/enrollment.rb [hunk 5]
```diff
@@ -304,6 +312,22 @@ def notify_administrators_for_unknown_software_enrollment
...
+    EnrollmentMailer.with(
+      target_api: target_api,
+      enrollment_id: id,
+      demandeur_email: demandeurs.pluck(:email).first
+    ).notify_instructors_submitted_enrollment.deliver_later
+  end
+
+  def notify_subscribers_for_submitted_enrollment_after_request_changes
```
Comment (rdubigny) [inline backend/app/models/enrollment.rb:323]:
```suggestion
  def notify_subscribers_for_enrollment_submission_after_changes_requested
```

File: backend/app/models/enrollment.rb [hunk 6]
```diff
@@ -304,6 +312,22 @@ def notify_administrators_for_unknown_software_enrollment
...
+    EnrollmentMailer.with(
+      target_api: target_api,
+      enrollment_id: id,
+      demandeur_email: demandeurs.pluck(:email).first
+    ).notify_instructors_submitted_changes_requested.deliver_later
```
Comment (rdubigny) [inline backend/app/models/enrollment.rb:328]:
```suggestion
    ).submission_after_changes_requested_notification_email.deliver_later
```

**Review by rdubigny — CHANGES_REQUESTED** [review]
thanks for all the tests!

The implementation is OK, I got a few remarks on variable naming.

File: backend/config/data_providers.yml (hunk 1 above)
Comment (Isalafont) [reply backend/config/data_providers.yml:14]: I totally forgot to check the ReadMe ! Thank you

**Comment by Isalafont** [comment]
Thank you for your name suggestion, I've done the changes.

**Review by rdubigny — APPROVED** [review]

**Pull request merged**